packages/manager/apps/dedicated-servers/src/hooks/useLinkUtils.tsx

```diff
@@ -0,0 +1,27 @@
+import { useContext, useEffect, useState } from 'react';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+
+export type LinkEntry = { [key: string]: string | LinkEntry };
+export type UrlLinks = { [key: string]: LinkEntry | UrlEntry | string };
+export type UrlEntry = { [key: string]: string };
+
+function useLinkUtils<T extends string | UrlEntry | UrlLinks | LinkEntry>(
```

Comment:
Why is it so complicated ?

This hook just takes an object containing all links for all subs, get the subsidiary from the environment (which you can do without an useEffect by the way) and return the object containing the links for that sub.